numpy/lib/_function_base_impl.pyi

```diff
@@ -303,6 +303,16 @@ def diff(
     append: ArrayLike = ...,
 ) -> NDArray[Any]: ...
 
+@overload
+def interp(
+    x: _FloatLike_co,
+    xp: _ArrayLikeFloat_co,
+    fp: _ArrayLikeFloat_co,
+    left: None | _FloatLike_co = ...,
+    right: None | _FloatLike_co = ...,
+    period: None | _FloatLike_co = ...,
+) -> float: ...
```

Comment:
This return type technically isn't wrong, but could be tightened up by changing it into `float64` instead.

```suggestion
@overload
def interp(
    x: _FloatLike_co,
    xp: _ArrayLikeFloat_co,
    fp: _ArrayLikeFloat_co,
    left: None | _FloatLike_co = ...,
    right: None | _FloatLike_co = ...,
    period: None | _FloatLike_co = ...,
) -> float64: ...
```